tests/unit/test_main.cpp

```diff
@@ -1,2 +1,7 @@
 #define CATCH_CONFIG_MAIN
+#define CATCH_CONFIG_ENABLE_BENCHMARKING
 #include "catch2/catch.hpp"
+
+TEST_CASE("type tests", "[system]") {
+    REQUIRE(sizeof(uint64_t) == 8);
```

Comment:
probably also add sizeof(char *) == 4 to make sure the tests only run in 32bit mode

Reply:
Good opinion. Unfortunately I found out that testcases in this file run with every testfile due to linkage dependency of CATCH_CONFIG_MAIN. I will make separate system_tests.cpp where we will place such a useful type size tests.